### ISSUE TITLE
Load individual templates asynchronously

### ISSUE THREAD

**Issue by bluefire2121** [tinymce/tinymce#5637]
I'd like to get the contents of a template using an async fetch() call, but I can't seem to find an event associated with the template select dropdown item change. 

https://www.tiny.cloud/docs/plugins/template/

I was thinking of creating an event listener on the select dropdown list, but the Insert Template dialog is rendered dynamically. 

How do I monitor the change event of select dropdown list? I'd rather not get the contents of a template until it's needed.

Thanks

**Comment by lnewson** [comment]
Hi @bluefire2121,

You shouldn't be trying to change or listen for an event when an item is selected, as that's the plugin's job to fetch the data from the settings you provide. If you want to use async templates there's an undocumented way to do that by providing a callback function to the `templates` setting. Here's a fiddle to show you an example: http://fiddle.tinymce.com/Whhaab

Does that answer your question?

**Comment by bluefire2121** [comment]
Hi @lnewson,

Almost. Sending a function to templates would still return an array of objects all at once. 

What I'd like to get is the content of any given template on demand. 

I have 1000s of templates. I have two separate API calls. 

The first call is to get the list of template names. This is so the dropdown box inside the dialog is populated with results. 

I'd like to have a second, separate call to get the content of the value selected so I don't have to pull the content of every template initially (especially if most templates won't be viewed by the user). 

Is that a possibility? 

Thank you

**Comment by lnewson** [comment]
Yeah, that makes sense I can see why you'd want that. Unfortunately, I'm afraid that's not currently possible sorry as the template plugin only allows loading individual templates from a URL or a hardcoded string. I'll mark this as a feature request for our Product Management team to evaluate.

**Comment by bluefire2121** [comment]
Thank you for marking this a feature request. :-)

**Comment by NicolasCARPi** [comment]
Hello,

I came across this issue after realizing that the `templates` setting could be an `Array`, an URL to get the JSON, but unfortunately not a function. Thanks to the comment of @lnewson, I was able to make it work! Thanks Lee :)

Would it make sense to make a PR to the doc to document this very useful callback possibility? It's a pity to have a perfectly working functionality that needs inner knowledge to use! ;)
